astropy/table/table.py

```diff
@@ -1541,13 +1543,53 @@ def add_column(self, col, index=None, rename_duplicate=False):
               2 0.2 1.2
               3 0.3 1.3
 
+        Add an unnamed column or mixin object at the end of the table using a default name::
+
+            >>> t = Table([[1, 2, 3], [0.1, 0.2, 0.3]], names=('a', 'b'))
+            >>> col_c = Column(data=['x', 'y', 'z'])
+            >>> t.add_column(col_c)
+            >>> print(t)
+             a   b   2
+            --- --- ---
+              1 0.1   x
+              2 0.2   y
+              3 0.3   z
+
+        Add an unnamed column or mixin object at the end of the table by specifying an explicit name::
+
+            >>> t = Table([[1, 2, 3], [0.1, 0.2, 0.3]], names=('a', 'b'))
+            >>> col_c = Column(data=['x', 'y', 'z'])
+            >>> t.add_column(col_c, name='c')
+            >>> print(t)
+              a   b  c
+            --- --- ---
+              1 0.1   x
+              2 0.2   y
+              3 0.3   z
+
+        Add a column or mixin object at the end of the table and specify a different name::
+
+            >>> t = Table([[1, 2, 3], [0.1, 0.2, 0.3]], names=('a', 'b'))
+            >>> col_b = Column(name='b', data=[1.1, 1.2, 1.3])
+            >>> t.add_column(col_b, name='c')
+            >>> print(t)
+             a   b   c
+            --- --- ---
+              1 0.1 1.1
+              2 0.2 1.2
+              3 0.3 1.3
+
         To add several columns use add_columns.
         """
         if index is None:
             index = len(self.columns)
-        self.add_columns([col], [index], rename_duplicate=rename_duplicate)
+        if name is None:
```

Comment:
I think here you do not actually need to set the default name, but rather just pass on `None`, since `add_columns` will do the right thing. Instead, you'd need
```
if name is not None:
   name = (name,)
```